sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -37,6 +38,14 @@ private[sql] object Column {
   def apply(expr: Expression): Column = new Column(expr)
 
   def unapply(col: Column): Option[Expression] = Some(col.expr)
+
+  private[sql] def generateAlias(e: Expression, index: Int): String = {
+    e match {
+      case a: AggregateExpression if a.aggregateFunction.isInstanceOf[TypedAggregateExpression] =>
+        s"${a.aggregateFunction.prettyName}_c${index}"
```

Comment:
how about `aggregateFunction.toString`? It carries more information and not that verbose.